[WebGPU] Pipeline layout for compute pipelines is not validated to be valid

mwyrzykowski opened the pull request:
#### 2c15e1198606b092facb2395ba04471e7fcd53a7
<pre>
[WebGPU] Pipeline layout for compute pipelines is not validated to be valid
<a href="https://bugs.webkit.org/show_bug.cgi?id=286250">https://bugs.webkit.org/show_bug.cgi?id=286250</a>
<a href="https://rdar.apple.com/143249146">rdar://143249146</a>

Reviewed by Tadeu Zagallo.

Unlike RenderPipelines, creating a ComputePipeline did not check if the
associated PipelineLayout was invalid. Attempting to do so should result
in an invalid ComputePipeline according to the WebGPU specification, see
<a href="https://www.w3.org/TR/webgpu/#dom-gpudevice-createcomputepipeline">https://www.w3.org/TR/webgpu/#dom-gpudevice-createcomputepipeline</a>

* LayoutTests/fast/webgpu/regression/repro_1380-expected.txt: Added.
* LayoutTests/fast/webgpu/regression/repro_1380.html: Added.
* LayoutTests/fast/webgpu/regression/repro_1380b-expected.txt: Added.
* LayoutTests/fast/webgpu/regression/repro_1380b.html: Added.
* LayoutTests/fast/webgpu/regression/repro_1380c-expected.txt: Added.
* LayoutTests/fast/webgpu/regression/repro_1380c.html: Added.
* LayoutTests/fast/webgpu/regression/repro_1380d-expected.txt: Added.
* LayoutTests/fast/webgpu/regression/repro_1380d.html: Added.
Add associated regression tests and expectations.

* Source/WebGPU/WebGPU/ComputePipeline.mm:
* Source/WebGPU/WebGPU/IsValidToUseWith.h:
(WebGPU::isValidToUseWithDevice):
* Source/WebGPU/WebGPU/RenderPipeline.mm:

Canonical link: <a href="https://commits.webkit.org/289162@main">https://commits.webkit.org/289162@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/b6620cc5e1ccb31da0aac4f961e094727531ff83

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/85501 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/5208 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/39909 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/90602 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/36516 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/5381 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/13188 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/66451 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/24264 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/88522 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/4127 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/77635 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/46733 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/4006 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/31897 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/35585 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/74693 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/32747 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/92206 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/12828 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/9404 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/75129 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/13049 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/73468 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/74259 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/18356 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/18536 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/16970 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/4915 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/12833 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/18234 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/12643 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/16086 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/14413 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

tadeuzagallo reviewed (approved):
Just a nit: the title is a bit confusing, but I get what it means

mwyrzykowski committed:
[WebGPU] Pipeline layout for compute pipelines is not validated to be valid
https://bugs.webkit.org/show_bug.cgi?id=286250
rdar://143249146

Reviewed by Tadeu Zagallo.

Unlike RenderPipelines, creating a ComputePipeline did not check if the
associated PipelineLayout was invalid. Attempting to do so should result
in an invalid ComputePipeline according to the WebGPU specification, see
https://www.w3.org/TR/webgpu/#dom-gpudevice-createcomputepipeline

* LayoutTests/fast/webgpu/regression/repro_1380-expected.txt: Added.
* LayoutTests/fast/webgpu/regression/repro_1380.html: Added.
* LayoutTests/fast/webgpu/regression/repro_1380b-expected.txt: Added.
* LayoutTests/fast/webgpu/regression/repro_1380b.html: Added.
* LayoutTests/fast/webgpu/regression/repro_1380c-expected.txt: Added.
* LayoutTests/fast/webgpu/regression/repro_1380c.html: Added.
* LayoutTests/fast/webgpu/regression/repro_1380d-expected.txt: Added.
* LayoutTests/fast/webgpu/regression/repro_1380d.html: Added.
Add associated regression tests and expectations.

* Source/WebGPU/WebGPU/ComputePipeline.mm:
* Source/WebGPU/WebGPU/IsValidToUseWith.h:
(WebGPU::isValidToUseWithDevice):
* Source/WebGPU/WebGPU/RenderPipeline.mm:

Canonical link: https://commits.webkit.org/289162@main
